projects/igniteui-angular/src/lib/select/select.component.ts

```diff
@@ -57,12 +75,28 @@ const noop = () => { };
         }
     `]
 })
-export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelectBase, ControlValueAccessor, AfterContentInit,
-    EditorProvider {
+export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelectBase, ControlValueAccessor,
+    AfterContentInit, OnInit, AfterViewInit, OnDestroy, EditorProvider {
 
     /** @hidden @internal do not use the drop-down container class */
     public cssClass = false;
 
+    /**
+     * @hidden @internal
+     */
+    @HostBinding('class.igx-input-group--valid')
```

Comment:
Not super sure the input group classes should be bound to the select. While this currently works, because the rule is not restrictive for the structure, it may very well be in the future or the class name can change.
Plus, it doesn't make sense semantically to have it on the host.

As for alternatives - you _could_ move this to the group or better yet - I see a `valid` setter on the input.